### PR TITLE
Refactor go-crypto dependencies to avoid C secp256k1 lib

### PR DESCRIPTION
**Pull request by Schwartz10**
Note that I am not a cryptographer, and this refactor should be reviewed in depth by someone with deeper domain knowledge. A couple things to note:

- Certain calls handled previously by the C library panic instead of throwing errors. I've imitated that functionality in go-crypto methods that do not return errors 
- I'm pretty confident in the `v` bit logic and all tests pass, but the compact / compact recoverable signature / verification logic should be reviewed for correctness and edge cases 
- There wasn't a good drop in replacement for `secp256k1.S256()` method, so i brought in the drec lib which seemingly has a drop in replacement. If the reviewers like the logic in this PR, i have no problem following up with copy/pasting the necessary code / lisences directly into this repo

### CODE REVIEW

File: crypto.go [hunk 1]
```diff
@@ -66,5 +87,31 @@ func GenerateKey() ([]byte, error) {
...
+
+		pk, err := secpEc.RecoverPublicKey(msg, r, s, v)
+		if err != nil {
+			return nil, err
+		}
+
+		return pk.Bytes(), nil
+	}
+
+	// parse the compact signature into its (`r`,`s`) scalars (without `v` recovery bit)
```
Comment (whyrusleeping) [inline crypto.go:98]: I feel like we probably want to handle other lengths explicitly?

Comment (Schwartz10) [reply crypto.go:98]: Wrt to just `EcRecover` or all methods that check signature lengths?

If all methods, were different sig lengths (other than 64, 65 bytes) handled prior to this refactor? 
- If yes, then i agree. 
- If no, then i would say handling other types of signatures should be a feature-add in a separate PR.
